src/components/RasterPreview.js

```diff
@@ -2,134 +2,171 @@
+    };
+  };
+
+  return (
+    <div
+      className={styles.MapContainer}
+    >
+      {chooseLocation ?
+        <div
+          className={styles.AssetSelect}
         >
-          <TileLayer
-            url={`https://api.mapbox.com/styles/v1/nelenschuurmans/ck8sgpk8h25ql1io2ccnueuj6/tiles/256/{z}/{x}/{y}@2x?access_token=${mapBoxAccesToken}`}
-            attribution="&copy; <a href=&quot;http://osm.org/copyright&quot;>OpenStreetMap</a> contributors"
-            setLocation={this.props.setLocation}
-          />
-          {raster ? (
-            <WMSTileLayer
-              url="/wms/"
-              styles={this.formatWMSStyles(raster.options.styles)}
-              layers={this.formatWMSLayers(raster.wms_info.layer)}
-              opacity={0.9}
-            />) : null}
-          <TileLayer
-            url={`https://api.mapbox.com/styles/v1/nelenschuurmans/ck8sgpk8h25ql1io2ccnueuj6/tiles/256/{z}/{x}/{y}@2x?access_token=${mapBoxAccesToken}`}
-            attribution="&copy; <a href=&quot;http://osm.org/copyright&quot;>OpenStreetMap</a> contributors"
+          <SelectDropdown
+            title={''}
+            name={name}
+            placeholder={'- Search and select an asset -'}
+            value={selectedAsset}
+            valueChanged={value => {
+              setSelectedAsset(value);
+              if (value) {
+                setLocation(value.value);
+              } else {
+                setLocation(null);
+              };
+            }}
+            options={[]}
+            validated={validated}
+            errorMessage={errorMessage}
+            triedToSubmit={triedToSubmit}
+            isAsync
+            loadOptions={searchInput => fetchAssets(raster, searchInput)}
           />
-          {location ?
-           <Marker position={marker} />
-          : null}
-        </Map>
-      </div>
-    );
-  }
+        </div>
+      : null}
+      <Map
+        onClick={handleMapClick}
+        className={styles.MapStyle}
+        zoomControl={false}
+        {...mapLocation}
+      >
+        <ZoomControl position="bottomright"/>
+        <TileLayer
+          url={`https://api.mapbox.com/styles/v1/nelenschuurmans/ck8sgpk8h25ql1io2ccnueuj6/tiles/256/{z}/{x}/{y}@2x?access_token=${mapBoxAccesToken}`}
+          attribution="&copy; <a href=&quot;http://osm.org/copyright&quot;>OpenStreetMap</a> contributors"
+          setLocation={setLocation}
+        />
+        {raster ? (
+          <WMSTileLayer
+            url="/wms/"
+            // @ts-ignore
+            styles={formatWMSStyles(raster.options.styles)}
+            layers={formatWMSLayers(raster.wms_info.layer)}
+            opacity={0.9}
+          />) : null}
+        <TileLayer
+          url={`https://api.mapbox.com/styles/v1/nelenschuurmans/ck8sgpk8h25ql1io2ccnueuj6/tiles/256/{z}/{x}/{y}@2x?access_token=${mapBoxAccesToken}`}
+          attribution="&copy; <a href=&quot;http://osm.org/copyright&quot;>OpenStreetMap</a> contributors"
+        />
```

Comment:
I do not understand why a second tileLayer is needed here.
Could you explain it with a comment?

Reply:
Also kept from the old code. It's a duplicate of the previous tile layer I guess so I removed it too.